src/lib.rs

```diff
@@ -270,24 +291,31 @@ impl<T: AsRef<str>> fmt::Display for Grid<T> {
         // We overestimate how many spaces we need, but this is not
         // part of the loop and it's therefore not super important to
         // get exactly right.
-        let padding = " ".repeat(self.widest_cell_width);
+        let padding = " ".repeat(self.widest_cell_width + self.options.filling.width());
 
         for y in 0..self.dimensions.num_lines {
+            // Current position on the line.
+            let mut cursor: usize = 0;
             for x in 0..self.dimensions.widths.len() {
-                let num = match self.options.direction {
-                    Direction::LeftToRight => y * self.dimensions.widths.len() + x,
-                    Direction::TopToBottom => y + self.dimensions.num_lines * x,
+                let (num, next) = match self.options.direction {
```

Comment:
So `num` here was already not very descriptive (which wasn't your fault obviously), but with `next` it even more unclear. Is it possible to make these names a bit more descriptive? Currently, it's not super clear that `num` is an absolute value, but `next` is an offset.

Maybe some of the explanation above `last_in_row` could go here.